ui/src/app/apidocs/components/apiDocs.tsx

```diff
@@ -1,13 +1,20 @@
 import {Page} from 'argo-ui';
 import * as React from 'react';
+import {useEffect} from 'react';
 import SwaggerUI from 'swagger-ui-react';
 import 'swagger-ui-react/swagger-ui.css';
 import {uiUrl} from '../../shared/base';
+import {services} from '../../shared/services';
 
-export const ApiDocs = () => (
-    <Page title='Swagger'>
-        <div className='argo-container'>
-            <SwaggerUI url={uiUrl('assets/openapi-spec/swagger.json')} defaultModelExpandDepth={0} />
-        </div>
-    </Page>
-);
+export const ApiDocs = () => {
+    useEffect(() => {
```

Comment:
could we wrap this. func up in, say `useCollectEvent('openedApiDocs');`, if that does not work, then `useEffect(() => collectEvent('openedApiDocs'))`

Reply:
I think we can wrap that where `useEffect` is used, but what should we do where it's used in class like [here](https://github.com/argoproj/argo-workflows/blob/bd1c2baf57083a0c2b35be8c111e0620bec3ad43/ui/src/app/workflows/components/workflows-list/workflows-list.tsx#L126)?
Should we leave them as it is and just wrap it for `useEffect` use case?

Reply:
Use both?

Reply:
Ah OK, I'm sorry for the confusion.